src/net/sourceforge/kolmafia/persistence/ItemFinder.java

```diff
@@ -277,7 +278,8 @@ private static void filterNameList(List<String> nameList, Match filterType) {
           itemId != -1
               && !ItemDatabase.getAttribute(
                   itemId, EnumSet.of(Attribute.TRADEABLE, Attribute.MATCHABLE, Attribute.QUEST))
-              && !NPCStoreDatabase.contains(itemId));
+              && !NPCStoreDatabase.contains(itemId)
+              && !InventoryManager.itemAvailable(itemId));
```

Comment:
This doesn't only check for items in your inventory, as the comment says (that would be `hasItem`).

Previous discussion at https://kolmafia.us/threads/change-item-string-fuzzy-matching-to-prefer-owned-items.29044/#post-172700 (I think this is the same thing).

Given

```
    // Never match against (non-quest) untradeable items not available
    // in NPC stores when other items are possible.
```

I am curious about why we are doing this at all. Though given
```
    // If there are multiple matches, such that one is a substring of the
    // others, choose the shorter one, on the grounds that the user would have
    // included part of the unique section of the longer name if that was the
    // item they actually intended.  This makes it easier to refer to
    // non-clockwork in-a-boxes, and DoD potions by flavor.
```
elsewhere in the file, how about adding "matchable" to the non-replica items? I think that would prefer the non-replicas to the replicas.